web/src/gear_menu.js

```diff
@@ -161,22 +162,16 @@ export function initialize() {
             // it being called before message lists are initialized and the order cannot be changed.
             // Also, since these buttons are only visible for spectators which doesn't have events,
             // if theme is changed in a different tab, the theme of this tab remains the same.
-            $popper.on("click", "#gear-menu-dropdown .gear-menu-select-dark-theme", (e) => {
-                popover_menus.hide_current_popover_if_visible(instance);
-                e.preventDefault();
-                e.stopPropagation();
-                requestAnimationFrame(() => {
-                    dark_theme.enable();
-                    message_lists.update_recipient_bar_background_color();
-                });
-            });
-
-            $popper.on("click", "#gear-menu-dropdown .gear-menu-select-light-theme", (e) => {
-                popover_menus.hide_current_popover_if_visible(instance);
-                e.preventDefault();
-                e.stopPropagation();
+            $popper.on("change", "input[name='theme-select']", (e) => {
+                const theme_code = Number.parseInt($(e.currentTarget).attr("data-theme-code"), 10);
                 requestAnimationFrame(() => {
-                    dark_theme.disable();
+                    if (theme_code === settings_config.color_scheme_values.night.code) {
+                        dark_theme.enable();
+                    } else if (theme_code === settings_config.color_scheme_values.day.code) {
+                        dark_theme.disable();
+                    } else {
+                        dark_theme.default_preference_checker();
+                    }
                     message_lists.update_recipient_bar_background_color();
```

Comment:
Let's move this call inside `dark_theme` and calling it in the server events dispatch path too.

Reply:
Moved the CSS class logic into `dark_theme.set_theme()` and added `dark_theme.set_theme_and_update()`, which calls `set_theme()` and also calls the realm logo and recipient bar update methods. The idea is that all theme switchers and commands call `dark_theme.set_theme_and_update()`, which will set the theme and also update the necessary elements. While `dark_theme.set_theme()` is used in `ui_init.js` to only set the theme via the CSS classes.